File: JVM/aliens-java/src/main/java/org/example/api/ISimultaneous.java
package org.example.api;

import java.util.concurrent.CompletableFuture;

import org.javatuples.Pair;

public interface ISimultaneous extends AutoCloseable {

    CompletableFuture<Integer> getId();

    CompletableFuture<Void> loadIt(String word);

    CompletableFuture<String> removeIt();

    CompletableFuture<Pair<Integer, Long>> runIt(int waitMs, int idx);
}

File: NET/Aliens.Sharp/API/ISimultaneous.cs
using System;
using System.Threading.Tasks;

namespace Example.API
{
    public interface ISimultaneous : IDisposable
    {
        Task<int> GetId();

        Task LoadIt(string word);

        Task<string> RemoveIt();

        Task<Tuple<int, long>> RunIt(int waitMs, int idx);
    }
}
